runtime/wasm/src/asc_abi/class.rs

```diff
@@ -176,15 +187,24 @@ impl AscType for AscString {
             asc_layout.push(high_byte);
         }
 
-        asc_layout
+        Ok(asc_layout)
     }
 
     /// The Rust representation of an Asc object as layed out in Asc memory.
-    fn from_asc_bytes(asc_obj: &[u8]) -> Self {
+    fn from_asc_bytes(asc_obj: &[u8]) -> Result<Self, DeterministicHostError> {
         // Pointer for our current position within `asc_obj`,
         // initially at the start of the content skipping `length`.
         let mut offset = size_of::<u32>();
 
+        // TODO: Verify that the length specified matches the byte length
```

Comment:
What's the situation with this TODO?

Reply:
The code as it was before (and as it is now) does not verify that the value of the first 4 bytes specifying the string length coincides with the length of `asc_obj`.

Reply:
Ah got it